scripts/docs-content-link-rewrites/rewrite-links-plugin.ts

```diff
@@ -1,245 +1,58 @@
-import path from 'path'
-import { VFile } from 'vfile'
 import { Definition, Link } from 'mdast'
 import { Plugin } from 'unified'
 import visit from 'unist-util-visit'
-import { productSlugsToHostNames } from 'lib/products'
+import { processDocsLinkNode } from 'lib/remark-plugins/remark-plugin-adjust-link-urls/helpers'
+import { rewriteTutorialsLink } from 'lib/remark-plugins/rewrite-tutorial-links/utils/rewrite-tutorials-link'
+import { getTutorialMap } from 'lib/remark-plugins/rewrite-tutorial-links/utils'
 
-const DOT_IO_HOSTNAMES = Object.values(productSlugsToHostNames)
-
-interface VFileData {
-	linksToRewrite: Record<string, string>
-	unrewriteableLinks: string[]
-}
-
-interface CustomVFile extends VFile {
-	data: VFileData
-}
-
-const getIsRelativeUrl = (url: string) => {
```

Comment:
❓ Trying to keep up with all the refactors 😅 Was this code thats being deleted refactored into the shared plugin? I remember reviewing test cases that seem similar to what these functions did. 🔪 🔪

Reply:
No worries! This code was totally deleted in favor of already existing code that lives in the two link rewriter plugins we use server-side when processing docs and tutorials content.

Reply:
Okay that makes sense. So this script essentially plays off of the refactor where you wanted to leverage the existing rewrite plugins, accepting a generic `urlAdjustFn` – which is presumably where the tutorial / devdot url mapping work occurs? And those functions passed in are the shared ones with the tutorials / docs rewrite plugins?

Reply:
Oh wait no I see the `processDocsNode` and the tutorial functions down below

Reply:
Okay so this just refactors to use those same functions which we run in the tutorials / docs plugins.

Reply:
Here's the recent PR where I abstracted the 2 plugin bodies into reusable functions (which I'm calling here in this refactor): https://github.com/hashicorp/dev-portal/pull/1530